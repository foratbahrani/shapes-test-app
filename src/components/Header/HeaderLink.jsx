import React from 'react';
import { Link, withRouter } from 'react-router-dom';

export const HeaderLink = ({ to, children, history }) => {
	const { pathname } = history.location;
	const getClassName = () => (pathname.includes(to) ? 'active' : '');

	return (
		<Link to={to} className={getClassName()}>
			{children}
		</Link>
	);
};

export default withRouter(HeaderLink);
