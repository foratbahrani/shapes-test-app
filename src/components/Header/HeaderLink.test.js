import { shallow } from 'enzyme';
import React from 'react';
import { HeaderLink } from './HeaderLink';

it('matches snapshot', () => {
	const history = { location: { pathname: '' } };
	expect(shallow(<HeaderLink to='' history={history} />)).toMatchSnapshot();
});

it('activates when needed', () => {
	const history = { location: { pathname: '/test' } };
	expect(shallow(<HeaderLink to='/test' history={history} />).props().className).toEqual('active');
});

it('deactivates when needed', () => {
	const history = { location: { pathname: '/meh' } };
	expect(shallow(<HeaderLink to='/test' history={history} />).props().className).toEqual('');
});
