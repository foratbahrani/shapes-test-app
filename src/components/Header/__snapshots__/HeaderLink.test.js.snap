// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Link
  className="active"
  to=""
/>
`;
